lldb/source/Plugins/LanguageRuntime/Swift/SwiftLanguageRuntimeNames.cpp

```diff
@@ -730,6 +727,17 @@ SwiftLanguageRuntime::DemangleSymbolAsString(StringRef symbol, DemangleMode mode
   return swift::Demangle::demangleSymbolAsString(symbol, options);
 }
 
+swift::Demangle::NodePointer
+SwiftLanguageRuntime::DemangleSymbolAsNode(llvm::StringRef symbol,
+                                           swift::Demangle::Context *ctx) {
+  LLDB_LOGF(GetLog(LLDBLog::Demangle), "demangle swift as node: '%s'",
+            symbol.str().data());
+  if (ctx)
+    ctx->demangleSymbolAsNode(symbol);
+  Context local_ctx;
+  return local_ctx.demangleSymbolAsNode(symbol);
```

Comment:
The context owns the nodes, so won't this return garbage in the `local_ctx` case?

Reply:
🤦 good point I clearly did this without thinking through

Reply:
fixed, thanks!